streams/sensitive-data.js

```diff
@@ -4,12 +4,34 @@ const DEFAULT_SENSITIVE_DATA_FRAGMENTS =
   '(mdp|password|authorization|token|pwd|auth)';
 
 module.exports = class SensitiveDataStream {
-  constructor(fragments) {
+  constructor(fragments, patterns = []) {
     this.fragments = fragments || DEFAULT_SENSITIVE_DATA_FRAGMENTS;
-    this.pattern = new RegExp(`"${this.fragments}":"([^"]*)"`, 'ig');
+    this.replacer = '__SENSITIVE_DATA__';
+
+    // If a pattern is provided
+    if (patterns.length) {
+      this.patterns = patterns;
+    } else {
+      // If no pattern provided, then add 2 default regexes
+      this.patterns = [
+        {
+          regex: new RegExp(`"${this.fragments}":"([^"]*)"`, 'ig'), // @Match "mdp":"My super password"
+          substitute: `"$1":"${this.replacer}"`,
+        },
+        {
+          regex: new RegExp(`${this.fragments}=([\\w-]*)`, 'ig'), // @Match mdp=My-super-password
+          substitute: `$1=${this.replacer}`,
+        },
+      ];
+    }
   }
   write(input) {
-    const sanitized = input.replace(this.pattern, '"$1":"__SENSITIVE_DATA__"');
+    let sanitized = input;
+
+    // Apply replace on input looping through patterns array
+    for (let pattern of this.patterns) {
```

Comment:
All the logging process is done synchronously, we must ensure this array has not too many items because it could lead to performance issues. 
If we don't do anything in code we must at least warn the users in the documentation.

Reply:
I just updated README because for the moment there is no usage of this functionality. But if you prefer I can update code.